<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
</head>
<body>
    <h1>Fetch Data by Using Promise</h1>
    <ul>
        <p>Fetching 5 todos...<br> Each one takes 1 second </p>
    </ul>

<script
    src="https://code.jquery.com/jquery-3.4.1.min.js"
    integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo="
    crossorigin="anonymous"></script>
<script>

function  getTodosUsingPromise(number){
    return new Promise((resolve, reject) => {
        $.ajax({
            url: `https://jsonplaceholder.typicode.com/todos/${number}`,
            method: "GET",
            success: (data) => setTimeout(() => resolve(data), 1000),
            error: (err) => setTimeout(() => reject(data), 1000)
        })
    })
}


async function init(){
    console.log('start');
    let listHtml = ''; 

    try{
        for(let i = 1; i <= 5; i++){
            const data = await getTodosUsingPromise(i);      
            //console.log(data);
            listHtml += `
            <li>
                <p>id: ${data.id}</p>
                <p>title: ${data.title}</p>
                <p>completed: ${data.completed}</p>
            </li>
            `;
        }
    }catch(err) {
        listHtml += `
          <li>
              <p>id: not found</p>
              <p>title: not found</p>
              <p>completed: not found</p>
          </li>
        `;
        console.log('not done, because',err.statusText);
    }
    document.querySelector('ul').innerHTML = listHtml;
    console.log('final');
}


init();


</script>    
</body>
</html>